### ISSUE TITLE
File not found error for "hh.hpp" during raftlib compilation using pkgconfig 

### ISSUE THREAD

**Issue by ameyand98** [RaftLib/RaftLib#152]
## Describe the bug
Raftlib code compilation gives "file does not exist error" for "raftinc/hh.hpp." On inspection, the file is included in defs.hpp but doesn't exist anywhere in the source code. Not sure if its renamed or removed completely 

## Steps To Reproduce
* run any example (the sum example in the getting started section of raftlib), and it doesn't compile using pkg-config by throwing the error

## Expected behavior
Raftlib example code to compile

**Comment by jonathan-beard** [comment]
Ok, looks like I made a mistake with using a gist as a submodule. Let me see the best way to fix this. Test cases pass (which we need to fix) b/c cmake just skips over since it's not a fatal error. 

Message received is below, will update ticket when I have it resolved. 

```bash
INFO Initializing sub-module /RaftLib/git-dep/highwayhash from git repo!
fatal: remote error: upload-pack: not our ref fe62635a9c42eb310245d3bdbe3c05184f8d1089
fatal: the remote end hung up unexpectedly
Fetched in submodule path 'git-dep/highwayhash', but it did not contain fe62635a9c42eb310245d3bdbe3c05184f8d1089. Direct fetching of that commit failed.
-- Performing Test COMPILER_SUPPORTS_CXX17
-- Performing Test COMPILER_SUPPORTS_CXX17 - Success
```

**Comment by jonathan-beard** [comment]
Need to update header file copy routines so that ```hh.hpp``` ends up in the right place after install, then pig-config will be able to find without modification. Will push that next.

**Comment by jonathan-beard** [comment]
For now, the non-string names feature is a bit experimental. String names are the way to go so, let's not use ```-DSTRING_NAMES=0``` outside of the build tree. Once we're ready and that feature is suitably setup for prime time, we can add full support for external builds in PKG_CONFIG. I'm a bit hesitant now given all the test cases really need to be duplicated for that configuration. The reason I haven't yet, is, as it turns out string names for the most part are just as fast as integer ones. That being said, there were two issues, one where we had a gist download issue (fixed that), a second where the header file isn't installed from the highwayhash folder (have't fixed that yet), and another with defs.hpp on windows (fixed that one). Closing for now, will fix the install issue once we've finished duplicating a sufficient number of the test cases.
